Tests/Model/ApiPayloadTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace MauticPlugin\MauticContactClient\Tests\Model;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use MauticPlugin\MauticContactClientBundle\Entity\ContactClient;
+use MauticPlugin\MauticContactClientBundle\Services\Transport;
+
+class ApiPayloadTest extends MauticMysqlTestCase
+{
+    /** @test */
+    public function it_tokenizes_a_post_url_in_test_mode()
+    {
+        $container = [];
+        $history   = Middleware::history($container);
+        $stack     = HandlerStack::create();
+        // Add the history middleware to the handler stack.
+        $stack->push($history);
+        $this->container->set('mautic.contactclient.service.transport', new Transport(new Client(), $stack));
+
+        $apiPayload = $this->container->get('mautic.contactclient.model.apipayload');
+
+        $client = new ContactClient();
+        $client->setType('api');
+        $payload = $this->getPayload();
+        $client->setAPIPayload($payload);
+
+        $apiPayload->setTest(true)
+                   ->setContactClient($client);
+
+        $apiPayload->run();
+
+        foreach ($container as $item) {
+            /** @var Request $req */
+            $req = $item['request'];
+            $q   = $req->getUri()->getQuery();
+            $this->assertEquals('subid=&addr=1234%20Test%20St.&ad1=1234%20Test%20St.', $q);
+        }
+    }
+
+    private function getPayload()
+    {
+        $reflection = new \ReflectionClass($this);
+
+        $dir =  explode('/', $reflection->getFilename());
+        array_pop($dir);
+        $dir = implode('/', $dir);
```

Comment:
What the hell are you trying to do in here?

Reply:
Getting a premade API payload JSON to use within the test, since there's no class to generate one. 

It'd be nice if there was a class that could have methods or fields to work with instead of a raw json payload.

Reply:
That's option one.

Option 2 is to have a way to retrieve a path to Test Assets that you can reuse. 

Option 3, which I wouldn't suggest, but it's still there: Create a Class with the Test Data, pull it, and convert it to JSON.  
But don't do that )

Reply:
`__DIR__`?